api/bnd.bnd

```diff
@@ -2,9 +2,23 @@
     org.eclipse.microprofile.*
 
 Import-Package: \
-    javax.enterprise.util;version="[1.1,3)", \
+    javax.enterprise.*;-remove-attribute:=version, \
+    javax.inject;-remove-attribute:=version,\
     *
 
+Require-Capability: \
+    osgi.contract;\
+        osgi.contract=JavaCDI;\
+        filter:="(&(osgi.contract=JavaCDI)(version=${cdi.api.version}))",\
+    osgi.contract;\
+        osgi.contract=JavaInject;\
+        filter:="(&(osgi.contract=JavaInject)(version=${inject.api.version}))",\
+    osgi.serviceloader;\
+        filter:="(osgi.serviceloader=org.eclipse.microprofile.config.spi.ConfigProviderResolver)";\
+        cardinality:=multiple,\
+    osgi.extender;\
+        filter:="(osgi.extender=osgi.serviceloader.processor)"
```

Comment:
-1 on the changes. I disagree with the changes. The change will directly introduce dependencies on OSGi CDI, which will force the runtime to use service loader and OSGi CDI spec.

Reply:
This is the way one does OSGi these days. Better than having special methods in our API which are only needed for old OSGi containers and potentially dangerous for all others.

Reply:
we've applied this in a passive way using a _weak requirement_. See bellow.